Index: docroot/sites/all/modules/contrib/ddf/ddf.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- docroot/sites/all/modules/contrib/ddf/ddf.module	(revision bacb89fbdb66213b6a0c9be90440a07ac72f8b07)
+++ docroot/sites/all/modules/contrib/ddf/ddf.module	(revision )
@@ -365,6 +365,13 @@
     return $element;
   }
 
+  if($element['#multiple'] == true) {
+    $element['#attributes'] = array(
+      'multiple' => 'multiple',
+      'name' => $element['#name'].'[]',
+    );
+  }
+
   if ($form_state['process_input']) {
     $entity_type = $element['#entity_type'];
     $bundle = $element['#bundle'];
