diff --git a/handlers/draggableviews_handler_fieldapi.inc b/handlers/draggableviews_handler_fieldapi.inc
index 4bde559..bd3edaf 100644
--- a/handlers/draggableviews_handler_fieldapi.inc
+++ b/handlers/draggableviews_handler_fieldapi.inc
@@ -27,12 +27,16 @@ class draggableviews_handler_fieldapi extends draggableviews_handler {
     $form = array();
     $options = array('' => t('- Select -'));
 
-    // Check all the sortings added to a view. Hope there is
-    // better way to determine them. Need to research.
-    $sorts = $field->view->display_handler->display->display_options['sorts'];
+    // Check all the sortings added to a view.
+    // TODO: Research a better way to do this.
+    $sorts = isset($field->view->display_handler->display->display_options['sorts'])
+        ? $field->view->display_handler->display->display_options['sorts']
+        : array();
     // If no sorts available for current display, use sorts from default display.
     if (empty($sorts)) {
-      $sorts = $field->view->display['default']->display_options['sorts'];
+      $sorts = isset($field->view->display['default']->display_options['sorts'])
+          ? $field->view->display['default']->display_options['sorts']
+          : array();
     }
     foreach ($sorts as $sort_option) {
       $field_name = $sort_option['field'];
@@ -52,6 +56,7 @@ class draggableviews_handler_fieldapi extends draggableviews_handler {
     }
 
     // If options are empty, show warning message.
+    // Count will allows be at least 1 because we add '- Select -'.
     if (count($options) == 1) {
       $form['field_warning'] = array(
         '#markup' => '<div class="messages warning">' . t('Add weight integer field to sorts so it can be selected.') . '</div>',
